Libplanet.Tests/Blocks/BlockMetadataTest.cs

```diff
@@ -360,28 +360,36 @@ public void CancelMineNonce(int? workers)
                     txHash: Block1Metadata.TxHash);
 
                 Exception exception = null;
+                Nonce? nonce = null;
+                ImmutableArray<byte>? hash = null;
                 Task task = Task.Run(() =>
                 {
                     try
                     {
                         if (workers is int w)
                         {
-                            Block1Metadata.MineNonce(w, source.Token);
+                            (nonce, hash) = metadata.MineNonce(w, source.Token);
                         }
                         else
                         {
-                            Block1Metadata.MineNonce(source.Token);
+                            (nonce, hash) = metadata.MineNonce(source.Token);
                         }
                     }
                     catch (OperationCanceledException ce)
                     {
                         exception = ce;
                     }
                 });
-
                 source.Cancel();
                 bool taskEnded = task.Wait(TimeSpan.FromSeconds(10));
                 Assert.True(taskEnded);
+                if (!(hash is null))
+                {
+                    Assert.True(ByteUtil.Satisfies(hash, long.MaxValue));
+                }
+
+                Assert.Null(nonce);
+                Assert.Null(hash);
```

Comment:
I think we can get rid of `if (!(hash is null))`? If `hash` is not null, the test won't pass anyway due to `Assert.Null(hash)` statement.

Reply:
What I've thought was,
"If mining has not been correctly canceled, nonce will exist, and we have to know if generated nonce matches the given difficulty."
Actually, at the first time mining wasn't canceled, and some nonce was generated.
But I couldn't know why mining with maximum difficulty successes within short time, so I've examined if given difficulty properly works, and found difficulty wasn't properly passed.

I hesitated to remove those statements because still this can give some evidence, but this idea can be biased. I'll hear your opinion.

Reply:
Several things:
- `if` statement is a dead piece of code **when working as expected**.
- `Assert.Null()` already gives what is inside `nonce` and `hash` if one of them is not `null`.
- Testing the wrong behavior, i.e. whether returned value satisfies the difficulty, is not in the scope of the test.